swe_roe/ex2.c

```diff
@@ -0,0 +1,1224 @@
+static char help[] = "Partial 2D dam break problem.\n";
+
+#include <assert.h>
+#include <math.h>
+#include <petsc.h>
+#include <petscdmda.h>
+#include <petscmat.h>
+#include <petscts.h>
+#include <petscvec.h>
+
+#define RDyAlloc(size, result) PetscCalloc1(size, result)
+
+typedef struct _n_User *User;
+
+typedef struct {
+  PetscReal X[3];
+} RDyCoordinate;
```

Comment:
What about RDyPoint?

Reply:
Or maybe RDyNode? We should also store other attributes in the Node or Point type, such as boundary code to define boundary and outlet.